### PR TITLE
Create graphic object animation

### PR DESCRIPTION
**Pull request by HugoN1**
# Create an animate method in Renderer class to animate a Sprite

### Planned changes :
- Create animate() method in Renderer class

### Not Planned changes :
- fix m_pos type to std::pair<float, float> to match sf::Transformable::setPosition() parameters type
- Improve Renderer constructor to set m_texture and m_sprite
- Change the Game class

** A Tutorial of How to use animation method is in #88 **

### CODE REVIEW

**Review by AndreasLETANTER — APPROVED** [review]

**Review by AndreasLETANTER — APPROVED** [review]
Alt Text

File: gui/src/game/IGame/Game.hpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 #pragma once
 #include "IGame.hpp"
+#include "../IRenderer/Renderer.hpp"
```
Comment (lbarreteau) [inline gui/src/game/IGame/Game.hpp:10]: Rename the folder because IRender was deleted

File: gui/Makefile
Comment (lbarreteau) [inline gui/Makefile:1]: Remove -Isrc or add every folder in it

File: gui/src/game/IRenderer/Renderer.cpp [hunk 2]
```diff
@@ -7,10 +7,18 @@
 
 #include "Renderer.hpp"
 
-Renderer::Renderer(std::string t_textureFileName, std::pair<unsigned int, unsigned int> t_pos)
+Renderer::Renderer(std::string const &t_textureFileName, std::pair<float, float> t_pos, sf::IntRect t_textureRect)
 {
     m_textureFileName = t_textureFileName;
+    if (!m_texture.loadFromFile(t_textureFileName))
+        throw std::runtime_error("Cannot load texture");
+    m_textureSize = m_texture.getSize();
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/Renderer.cpp:15]: Check getSize please

File: gui/src/game/IRenderer/Renderer.hpp [hunk 3]
```diff
@@ -6,23 +6,28 @@
...
-        sf::IntRect getTextureRect(void) override;
-        sf::Sprite getSprite(void) override;
-        std::pair<unsigned int, unsigned int> getPos(void) override;
+        Renderer() = default;
+        Renderer(std::string const &t_textureFileName, std::pair<float, float> t_pos, sf::IntRect t_textureRect);
+        ~Renderer() = default;
+        void setTexture(sf::Texture t_texture);
+        void setSprite(sf::Sprite t_sprite);
+        void setTextureRect(sf::IntRect t_intRect);
+        void animate();
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/Renderer.hpp:20]: (void)

**Review by lbarreteau — CHANGES_REQUESTED** [review]
image

File: gui/Makefile
Comment (HugoN1) [reply gui/Makefile:1]: If we add a folder in it, -Isrc will not include the folder

**Review by lbarreteau — APPROVED** [review]

**Pull request merged**